docs/native-modules.md

```diff
@@ -312,21 +320,24 @@ This can be also tied in with C++/WinRT event handlers or `IAsyncOperation<T>` l
 
 ```cpp
     REACT_METHOD(GetString, L"getString");
-    void GetString(bool error, React::ReactPromise<React::JSValue>&& result) noexcept
+    void GetString(React::ReactPromise<std::string>&& result) noexcept
     {
       if (error) {
         result.Reject("Failure");
       } else {
         something.Completed([result] (const auto& operation, const auto& status) {
           // do error checking, e.g. status should be Completed
-          std::wstring wtext{operation.GetResults()};
-          result.Resolve(React::JSValue{ ConvertWideStringToUtf8String(text) });
+          winrt::hstring result{operation.GetResults()};
+          result.Resolve(winrt::to_string(result));
         });
       }
     }
 ```
+See [JavaScript and Windows Runtime strings](#javascript-and-windows-runtime-strings) for more details.
 
-`REACT_SYNC_METHOD` isn't supported in Debug when the JS engine is Chrome V8. You will get an exception that reads:
+The [`JSValue`](native-modules-jsvalue.md) type can be used when the API returns a JavaScript objects or takes JavaScript objects as input parameters.

```

Comment:
>The [](start = 0, length = 3)

I would rather say that the expectation is that the native module methods parameters are expected to be related to the problem domain. ReadValue and WriteValue function overrides can be implemented for custom types. In case if the strongly type approach is not possible, the developers can use the JSValue that is an immutable representation of a JSON-like objects.

Reply:
I would love to point to the ReadValue and WriteValue documentation but we don't mention them anywhere in the docs. I've filed https://github.com/microsoft/react-native-windows-samples/issues/282 to add these APIs to the docs.